File: frontend/src/apis/news.ts
import api from "./axios";

export interface NewsDetailData {
    news_id: number;
    img: string;
    title: string;
    content: string;
    company: string;
    author: string;
    date: string;
}

export const fetchNewsById = async (newsId: number): Promise<NewsDetailData> => {
    const response = await api.get<NewsDetailData>(`/news/${newsId}`);
    return response.data;
};

File: backend/src/main/java/chanmin/sisters/hackathon/entity/News.java
package chanmin.sisters.hackathon.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Setter
@Getter
@Table(name = "news")
public class News {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long news_id;

    private String img;
    private String title;

    @Column(columnDefinition = "TEXT") // 긴 본문 내용
    private String content;

    private String company;
    private String author;
    private LocalDate date;
}
